osidb/middleware.py

```diff
@@ -32,3 +35,44 @@ def __call__(self, request):
             _logger.info(f"ACLs set from middleware for user {request.user.username}")
             set_user_acls(request.user.groups.all())
         return self.get_response(request)
+
+
+def get_userstr(user):
+    userstr = str(user)
+    if getattr(user, "is_authenticated", False):
+        if getattr(user, "email", False):
+            userstr = user.email
+        elif getattr(user, "username", False):
+            userstr = user.username
+    return userstr
+
+
+class WSGIRequestWithHook(WSGIRequest):
+    """
+    Updates pghistory context when the request.user attribute is updated in order
+    for context to be set before DB User-model lookup happens. User is added to the request
+    by Django in middleware, but added to the request object in the view layer by
+    django-rest-framework.
+    """
+
+    def __setattr__(self, attr, value):
+        _value = get_userstr(value) if attr == "user" else value
+        return super().__setattr__(attr, _value)
```

Comment:
Why is this different than
https://github.com/RedHatProductSecurity/osidb/pull/912/files#diff-f8968d26d83a0fb2b740bd909eca8e653f90d02b763a7a7b1b77892e195ca29fR58
and what is the purpose of the two PRs?

Reply:
This is a mistake--I need to rebase again master, **then** the other PR's branch.

Originally, the purpose was to stagger their releases, though they are separate functionalities with separate tickets, we could combine them into one PR

Reply:
I think I could actually just ditch the other commits in this PR, I just had the others in there for local development reasons.